### PR TITLE
#168359024 Get all device API endpoint

### PR DESCRIPTION
**Pull request by okiti**
**What does this PR do?**

1. Gets all devices related to an account (hospital_admin, ward_user and nurse)

**What are the relevant pivotal tracker stories?**
[#168359024](https://www.pivotaltracker.com/story/show/168359024)

### CODE REVIEW

File: src/db/device/index.js [hunk 1]
```diff
@@ -2,16 +2,29 @@ import deviceModel from './model';
 
 /**
  * @description Gets a single device from the database
- * @param {object} deviceDetails  The data to be used as filter
- * @param {string} deviceDetails._id
- * @param {string} deviceDetails.hospitalId
- * @param {string} deviceDetails.wardId
+ * @param {object} deviceMatch  The data to be used as filter
+ * @param {string} deviceMatch._id
```
Comment (IAMOTZ) [inline src/db/device/index.js]: Explain the fields here

File: src/db/device/index.js [hunk 2]
```diff
@@ -2,16 +2,29 @@ import deviceModel from './model';
...
   const Model = deviceModel;
-  return Model.findOne(params);
+  return Model.findOne(deviceMatch);
+};
+
+/**
+ * @description Gets all devices matching the query parameter
+ * @param {object} deviceMatch  The data to be used as filter for getting all devices
+ * @param {string} deviceMatch.hospitalId The hospitalId will used as the filter
+ * if the hospital_admin request for a device
```
Comment (IAMOTZ) [inline src/db/device/index.js]: No need or extra details on who is making the request, the db module does not know about that

File: src/http/controllers/device.js [hunk 3]
```diff
@@ -1,12 +1,20 @@
 import deviceService from '../../services/device';
 import deviceValidation from '../validations/device';
 
+/**
+ * @description Controller for "update admin user" API operation
```
Comment (IAMOTZ) [inline src/http/controllers/device.js]: Copy cater 😄 . This ain't `update admin user` controller.

File: src/http/controllers/device.js [hunk 4]
```diff
@@ -19,6 +27,29 @@ const getSingleDevice = async (req, res) => {
   }
 };
 
+/**
+ * @description Controller for "update admin user" API operation
```
Comment (IAMOTZ) [inline src/http/controllers/device.js]: Ditto ...

File: src/http/controllers/device.js [hunk 5]
```diff
@@ -19,6 +27,29 @@ const getSingleDevice = async (req, res) => {
...
+ * @param {object} req Express request object
+ * @param {object} res Express response object
+ */
+const getAllDevice = async (req, res) => {
+  const { user, userType, log } = res.locals;
+  log.debug('Executing the getAllDevice controller');
+  try {
+    const device = await deviceService.getAllDevice(user, userType, log);
```
Comment (IAMOTZ) [inline src/http/controllers/device.js]: You are fetching multiple devices here so it should be `devices` not `device`.

File: src/services/device.js [hunk 6]
```diff
@@ -9,8 +9,9 @@ const getSingleDevice = async (deviceId, user, userType, log) => {
       hospitalId: userType === 'hospital_admin' ? user._id : user.hospitalId,
       wardId: userType === 'ward_user' ? user._id : user.wardId,
     };
+    const purifyDeviceMatch = JSON.parse(JSON.stringify(deviceMatch));
```
Comment (IAMOTZ) [inline src/services/device.js]: `purifyDevceMatch` 🤣

File: src/db/device/index.js [hunk 7]
```diff
@@ -2,16 +2,29 @@ import deviceModel from './model';
...
 /**
  * @description Gets a single device from the database
- * @param {object} deviceDetails  The data to be used as filter
- * @param {string} deviceDetails._id
- * @param {string} deviceDetails.hospitalId
- * @param {string} deviceDetails.wardId
+ * @param {object} deviceMatch  The data to be used as filter
+ * @param {string} deviceMatch._id
+ * @param {string} deviceMatch.hospitalId
+ * @param {string} deviceMatch.wardId
```
Comment (IAMOTZ) [inline src/db/device/index.js]: Please, explain these fields.

File: src/http/controllers/device.js (hunk 3 above)
Comment (IAMOTZ) [inline src/http/controllers/device.js]: Copy cater 😄 . This ain't `update admin user` controller.

File: src/db/device/index.js [hunk 8]
```diff
@@ -2,16 +2,29 @@ import deviceModel from './model';
...
+  return Model.findOne(deviceMatch);
+};
+
+/**
+ * @description Gets all devices matching the query parameter
+ * @param {object} deviceMatch  The data to be used as filter for getting all devices
+ * @param {string} deviceMatch.hospitalId The hospitalId will used as the filter
+ * if the hospital_admin request for a device
+ * @param {string} deviceMatch.wardId if the ward or nurse requests for a device
+ *  then both the hospitalId and the wardId will be used as filters
```
Comment (IAMOTZ) [inline src/db/device/index.js]: The comments here need some revision, the db module does not know who is making the request so details like `...if hte hospital_admin request for ...` is not supposed to be here.

File: src/services/device.js (hunk 6 above)
Comment (okiti) [reply src/services/device.js]: i was going to do 
`let deviceMatch = JSON.parse(JSON.stringify(deviceMatch));`
if you have a better name it is so much welcomed. 🤣🤣🤣🤣

File: src/http/controllers/device.js (hunk 4 above)
Comment (okiti) [reply src/http/controllers/device.js]: fixed

File: src/http/controllers/device.js (hunk 5 above)
Comment (okiti) [reply src/http/controllers/device.js]: fix

File: src/db/device/index.js (hunk 7 above)
Comment (okiti) [reply src/db/device/index.js]: okay. Fixed

File: src/http/controllers/device.js (hunk 3 above)
Comment (okiti) [reply src/http/controllers/device.js]: fixed

File: src/db/device/index.js (hunk 8 above)
Comment (okiti) [reply src/db/device/index.js]: fixed

File: src/services/device.js (hunk 6 above)
Comment (IAMOTZ) [reply src/services/device.js]: The name is okay, it's just kinda funny.

**Pull request merged**